src/Mod/Part/App/GeometryPyImp.cpp

```diff
@@ -217,7 +217,8 @@ Py::Boolean GeometryPy::getConstruction(void) const
 
 void  GeometryPy::setConstruction(Py::Boolean arg)
 {
-    getGeometryPtr()->Construction = arg;
+    if(getGeometryPtr()->getClassTypeId() != Part::GeomPoint::getClassTypeId())

```

Comment:
> It is under epigraph: a) Modify the api so that no unintended use is made.

This is again only true in the context of the sketcher. But there might be examples where it is useful (although I can't tell you one at the moment). So, I wouldn't restrict this inside the Geometry class but only there where it's needed, i.e. in the sketcher code.

> If I am not wrong (I think I remember it from an old discussion), there is no point in making a point of type construction.

Yes, in a sketch I don't know of an example where it would make sense to have a point as real geometry. If it's a "free" point I guess you can't create a wire from it and if it's a "bound" point then it's actually obsolete because the curve's points can be used instead.